### ISSUE TITLE
Ability to download all library and asset files for all available versions of the game

### ISSUE THREAD

**Issue by TheMCNerd2017** [PrismLauncher/PrismLauncher#79]
### Role

Being able to have a complete set of library and asset files.

### Suggestion

Add the ability to download library and asset files for all available versions of the game.

### Benefit

So that players can play any version of the game in the future without having to rely on Mojang/Microsoft.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_This is a repost of my suggestion from the old repository. I asked in the support channel if I should repost my suggestions currently in the old repository to the new one and was told to move to the new one._

I recently saw Meri's YouTube video urging Java Edition players to download all Minecraft Java Edition resource files if they can do a little bit of scripting, and thought that having the ability to do this in the launcher would be incredibly useful, especially since you would be able to play any version of Minecraft that was available at the time of downloading the resource files without having to rely on Mojang/Microsoft having the files available for download in the future.

Microsoft could decide to pull a bedrock at any time and make older versions of the game unavailable by removing their libraries and assets for any reason. They've done this in the past to other things like older Office versions by pulling their installation media and updates as well as older Windows versions by pulling the installation media, even from areas like VLSC.

I'm also aware that this feature would require a good chunk of storage space (over 25GB I think), however most people playing PC games have tons of storage space available already so this would not be a big issue.

**Comment by 9p4** [comment]
I don't think this is in the scope of this project. There already exist scripts online that do this, [like this (old) one](https://github.com/xtream1101/MinecraftArchive)

**Comment by TheMCNerd2017** [comment]
> I don't think this is in the scope of this project. There already exist scripts online that do this, [like this (old) one](https://github.com/xtream1101/MinecraftArchive)

All the scripts that I have used don't save the files in the structure that PrismLauncher expects (they save in a file structure that mirrors the Mojang's/Microsoft's CDNs), so you'd have to organize everything into the correct locations manually. Also, PrismLauncher always seems to download certain libraries and assets upon launching a version of Minecraft for the first time, even if they already exist on-disk in the correct locations and match the expected checksums present in the library and asset index files.
